mvc/src/main/java/nextstep/web/support/RequestMethod.java

```diff
@@ -1,5 +1,14 @@
 package nextstep.web.support;
 
+import java.util.Arrays;
+
 public enum RequestMethod {
-    GET, HEAD, POST, PUT, PATCH, DELETE, OPTIONS, TRACE
+    GET, HEAD, POST, PUT, PATCH, DELETE, OPTIONS, TRACE;
+
+    public static RequestMethod of(String method) {
+        return Arrays.stream(values())
+                .filter(requestMethod -> requestMethod.name().equalsIgnoreCase(method))
+                .findFirst()
+                .orElseThrow(IllegalArgumentException::new);
+    }
```

Comment:
```suggestion
    GET("GET"), HEAD("HEAD"), POST("POST"), PUT("PUT"),
    PATCH("PATCH"), DELETE("DELETE"), OPTIONS("OPTIONS"), TRACE("TRACE");

    private static final Map<String, RequestMethod> REQUEST_METHOD_MAP = new HashMap<>();

    static {
        for (RequestMethod requestMethod : values()) {
            REQUEST_METHOD_MAP.put(requestMethod.method, requestMethod);
        }
    }

    public final String method;

    RequestMethod(final String method) {
        this.method = method;
    }

    public static RequestMethod of(final String method) {
        return REQUEST_METHOD_MAP.get(method.toUpperCase(Locale.ROOT));
    }
```

위에서 이름이 같은 경우 `valueOf` 를 통한 방법이 있다고 말씀드렸는데
다른값으로 원하는 ENUM을 찾길 희망하실때는 다음과 같이 캐싱을 이용한다면, 
찾을때 `O(1)` 그리고 이름이 아닌 다른값으로도 원하는 값을 찾을수 있게되죠 !

Reply:
저의 의도는 valueOf로 충분히 표현될 수 있다고 판단해 valueOf를 채택했습니다!
캐싱해둘 Map을 생성해두는 것도 비용일테구요!
하나 궁금한 것은 method.toUpperCase() 만으로도 대문자로 변환이 가능한데, 왜 method.toUpperCase(Locale.ROOT)를 추천해주셨는지 궁금합니다!

Reply:
조엘이 말씀하신것처럼, `method.toUpperCase()` 으로도 동일하게 작동해요.
사실 큰이유는 없구요. **Locale.ROOT** 를 넣어주면 어떤 언어든 전부 대문자로 바꾸어주는 기능을해서 저걸로 그냥 넣었습니다 
사실 영어만 들어올테니 `toUpperCase()`만 해도 상관없어요 ㅎ_ㅎ;;

- https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#toUpperCase(java.util.Locale)